src/dask_awkward/layers/layers.py

```diff
@@ -44,7 +44,7 @@ def __repr__(self) -> str:
 
 
 class ImplementsIOFunction(Protocol):
-    def __call__(self, *args, **kwargs) -> AwkwardArray:
+    def __call__(self, *args, **kwargs):
```

Comment:
Why drop the return?

Reply:
Since `*args` and `**kwargs` are untyped we get a mypy error; it's something I'd like to revisit if we can better hint args and kwargs here, I just hadn't thought about it yet, but I supposed we could put `Any` for now